### PR TITLE
Backport: Blockout cleanup (#1055)

### PR DESCRIPTION
**Pull request by xavierh**
* Move views into views package

* Update xsd

* Don’t use zero line width block, when we can just use view. More efficient in render stage, and looks the same.

* Small fixes, things sonar will like

* Move button handler to its own file

* Update and increase documentation

* Remove unused class, and created constant for rectangle color

* extend Consumer<Button>

* Curly brace line

Conflicts:
	src/main/java/com/minecolonies/coremod/client/gui/WindowHutDeliveryman.java

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/blockout/controls/TextField.java [hunk 1]
```diff
@@ -315,10 +317,10 @@ public void onFocus()
         cursorBlinkCounter = 0;
     }
 
-    @Override
     /**
      * Draw itself at positions mx and my.
      */
+    @Override
     protected void drawSelf(final int mx, final int my)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/blockout/controls/TextField.java:324]: MAJOR The Cyclomatic Complexity of this method "drawSelf" is 21 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
MAJOR Refactor this method to reduce its Cognitive Complexity from 24 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 51 issues
* ![MAJOR][MAJOR] 22 major
* ![MINOR][MINOR] 28 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Pane.java#L19](https://github.com/Minecolonies/minecolonies/blob/77f59332cedb8e20fb2fa396a782e65936f24d2e/src/main/java/com/minecolonies/blockout/Pane.java#L19): class "Pane" has 47 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [PaneParams.java#L349](https://github.com/Minecolonies/minecolonies/blob/77f59332cedb8e20fb2fa396a782e65936f24d2e/src/main/java/com/minecolonies/blockout/PaneParams.java#L349): Assign this magic number 100 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [PaneParams.java#L349](https://github.com/Minecolonies/minecolonies/blob/77f59332cedb8e20fb2fa396a782e65936f24d2e/src/main/java/com/minecolonies/blockout/PaneParams.java#L349): Assign this magic number 100 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [PaneParams.java#L420](https://github.com/Minecolonies/minecolonies/blob/77f59332cedb8e20fb2fa396a782e65936f24d2e/src/main/java/com/minecolonies/blockout/PaneParams.java#L420): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [PaneParams.java#L439](https://github.com/Minecolonies/minecolonies/blob/77f59332cedb8e20fb2fa396a782e65936f24d2e/src/main/java/com/minecolonies/blockout/PaneParams.java#L439): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [PaneParams.java#L439](https://github.com/Minecolonies/minecolonies/blob/77f59332cedb8e20fb2fa396a782e65936f24d2e/src/main/java/com/minecolonies/blockout/PaneParams.java#L439): Assign this magic number 8 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [PaneParams.java#L444](https://github.com/Minecolonies/minecolonies/blob/77f59332cedb8e20fb2fa396a782e65936f24d2e/src/main/java/com/minecolonies/blockout/PaneParams.java#L444): Assign this magic number 255 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [PaneParams.java#L444](https://github.com/Minecolonies/minecolonies/blob/77f59332cedb8e20fb2fa396a782e65936f24d2e/src/main/java/com/minecolonies/blockout/PaneParams.java#L444): Assign this magic number 24 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [ButtonImage.java#L13](https://github.com/Minecolonies/minecolonies/blob/77f59332cedb8e20fb2fa396a782e65936f24d2e/src/main/java/com/minecolonies/blockout/controls/ButtonImage.java#L13): class "ButtonImage" has 38 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [ButtonImage.java#L563](https://github.com/Minecolonies/minecolonies/blob/77f59332cedb8e20fb2fa396a782e65936f24d2e/src/main/java/com/minecolonies/blockout/controls/ButtonImage.java#L563): Extract this nested ternary operation into an independent statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3358)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Raycoms** [comment]
:shipit: seems alright

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1163/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 50 issues
* ![MAJOR][MAJOR] 22 major
* ![MINOR][MINOR] 27 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Pane.java#L19](https://github.com/Minecolonies/minecolonies/blob/07d253fe7d2ca3f709ac2583b3f1becf3d24f527/src/main/java/com/minecolonies/blockout/Pane.java#L19): class "Pane" has 47 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [PaneParams.java#L349](https://github.com/Minecolonies/minecolonies/blob/07d253fe7d2ca3f709ac2583b3f1becf3d24f527/src/main/java/com/minecolonies/blockout/PaneParams.java#L349): Assign this magic number 100 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [PaneParams.java#L349](https://github.com/Minecolonies/minecolonies/blob/07d253fe7d2ca3f709ac2583b3f1becf3d24f527/src/main/java/com/minecolonies/blockout/PaneParams.java#L349): Assign this magic number 100 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [PaneParams.java#L420](https://github.com/Minecolonies/minecolonies/blob/07d253fe7d2ca3f709ac2583b3f1becf3d24f527/src/main/java/com/minecolonies/blockout/PaneParams.java#L420): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [PaneParams.java#L439](https://github.com/Minecolonies/minecolonies/blob/07d253fe7d2ca3f709ac2583b3f1becf3d24f527/src/main/java/com/minecolonies/blockout/PaneParams.java#L439): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [PaneParams.java#L439](https://github.com/Minecolonies/minecolonies/blob/07d253fe7d2ca3f709ac2583b3f1becf3d24f527/src/main/java/com/minecolonies/blockout/PaneParams.java#L439): Assign this magic number 8 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [PaneParams.java#L444](https://github.com/Minecolonies/minecolonies/blob/07d253fe7d2ca3f709ac2583b3f1becf3d24f527/src/main/java/com/minecolonies/blockout/PaneParams.java#L444): Assign this magic number 255 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [PaneParams.java#L444](https://github.com/Minecolonies/minecolonies/blob/07d253fe7d2ca3f709ac2583b3f1becf3d24f527/src/main/java/com/minecolonies/blockout/PaneParams.java#L444): Assign this magic number 24 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [ButtonImage.java#L13](https://github.com/Minecolonies/minecolonies/blob/07d253fe7d2ca3f709ac2583b3f1becf3d24f527/src/main/java/com/minecolonies/blockout/controls/ButtonImage.java#L13): class "ButtonImage" has 38 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [ButtonImage.java#L563](https://github.com/Minecolonies/minecolonies/blob/07d253fe7d2ca3f709ac2583b3f1becf3d24f527/src/main/java/com/minecolonies/blockout/controls/ButtonImage.java#L563): Extract this nested ternary operation into an independent statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3358)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Raycoms** [comment]
:shipit:

**Review by marchermans — APPROVED** [review]

**Pull request merged**